src/dns_operations/mod.rs

```diff
@@ -93,6 +96,7 @@ impl DnsOperations {
     pub fn delete_dns(&self,
                       long_name          : &String,
                       private_signing_key: &::sodiumoxide::crypto::sign::SecretKey) -> Result<::routing::structured_data::StructuredData, ::errors::DnsError> {
+        debug!("Deleting dns record ...");
```

Comment:
This is message for an overall summary:
Logging function entry points is practically trying to log execution path, which i think is too verbose to be useful .. Infact i would argue that logging execution path is never useful (you have dumps and back traces for that). Especially the way client module are designed, the error handling is always propagated to the caller, so you will get the entire back-trace at will at the app-side of things. What really need to be logged are things like blocking calls, event mechanisms etc., where the execution path may not return to the caller under certain circumstances (and indefinitely) and then it becomes useful to know where the path last traced to. So i would like this entire task to be refactored to reflect that. For other cases we have very robust error handling/propagation and test cases, so that makes logging in those scenarios very redundant and needless throttling of performance. So this task must be redone keeping that in mind.